ocs_ci/ocs/ocp.py

```diff
@@ -373,3 +373,41 @@ def switch_to_default_rook_cluster_project():
         bool: True on success, False otherwise
     """
     return switch_to_project(defaults.ROOK_CLUSTER_NAMESPACE)
+
+
+def get_all_nodes():
+    """
+    Get all cluster nodes
+
+    Returns:
+        list: Cluster node dictionaries
+
+    """
+    nodes_obj = OCP(kind='node')
+    return nodes_obj.get()['items']
+
+
+def wait_for_nodes_ready(num_of_nodes):
+    """
+    Wait until all nodes are in Ready status
```

Comment:
boto has waiters - It is efficient to use that call instead of sampler

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Waiter.InstanceRunning

Reply:
@vasukulkarni but here I'm checking the nodes from OCP side with 'oc get node -o yaml' to make sure all nodes reached Ready

Reply:
ok thanks, I see it,  OCP also has wait for function which could be used but either way looks fine.